crates/nu-cli/src/shell/completer.rs

```diff
@@ -28,6 +31,17 @@ impl NuCompleter {
             .map(|block| completion::engine::completion_location(line, &block.block, pos))
             .unwrap_or_default();
 
+        let matcher = nu_data::config::config(Tag::unknown())
+            .ok()
+            .and_then(|cfg| cfg.get("completion_style").cloned())
+            .and_then(|v| v.as_string().ok())
+            .unwrap_or_else(|| "".to_string());
```

Comment:
Sorry I didn't catch this the first time, but we already have `completion_type` nested under `line_editor` in the config. I'm thinking we may want to rename this, and perhaps list it under `line_editor`. How about:

```toml
[line_editor]
completion_type = "circular" # or
completion_type = "list"

completion_matches = "case-sensitive" # or
completion_matches = "case-insensitive" # or
completion_matches = "fuzzy" # someday! 🙂
```

Reply:
OK,

Perhaps completion_heuristic, or completion_method. Ugh, naming things, hey?

I didnt like completion_matcher, as it didnt seem to be the right type of describer. completion_matches seems to be similar to completion_matcher but I think the tense is wrong or something.

completion_engine (too broad I think)
completion_method
completion_heuristic
completion_match_method

Just a couple of ideas there. Maybe this should be with the relevant parties, being that it is the config namespace, it would be good to have a coherent naming scheme, and not stuck with anything that is annoying, and needs supporting in the future.

@fdncred any thoughts on this?

p.s. I only just renamed completion_matcher to completion_style in the last couple of commits, so you wouldnt have seen that before.

Reply:
OK, this should be namespaced under line_editor.

I will change this.

Reply:
Yeah, `completion_matches` was mostly a suggestion, but I'm game for anything that feels sufficiently different than `completion_type`.

I think `method` is too similar to `type`. I believe `heuristic` and `match_method` are both great, but the former may be a little too "computer science-y", so my vote is for `completion_match_method`.

> Ugh, naming things, hey?

Yes. Always 😂